<launch>
  <arg name="world_frame" default="ugr/car_odom" />
  <arg name="base_link_frame" default="ugr/car_base_link" />
  <arg name="slam_base_link_frame" default="ugr/slam_base_link" />
  <arg name="map_prefix" default="fastslam/global"/>

  <node pkg="slam_mcl" output="screen" type="slam_mcl" name="slam_mcl" respawn="False">
    # Parameters
    <param name="world_frame" value="$(arg world_frame)" />
    <param name="base_link_frame" value="$(arg base_link_frame)" />
    <param name="slam_base_link_frame" value="$(arg slam_base_link_frame)" />

    #Parameters from config yaml file
    <rosparam file="$(find slam_mcl)/config/mcl.yaml" command="load" />

    # Input/output remapping
    <remap from="/input/observations" to="ugr/car/observations/lidar" />
    <remap from="/input/map" to="ugr/car/map/$(arg map_prefix)" />
    <remap from="/output/odom" to="/ugr/car/odometry/filtered/mcl" />
  </node>
</launch>